remotes/handlers.go

```diff
@@ -56,11 +56,22 @@ func FetchHandler(ingester content.Ingester, fetcher Fetcher) images.HandlerFunc
 func fetch(ctx context.Context, ingester content.Ingester, fetcher Fetcher, desc ocispec.Descriptor) error {
 	log.G(ctx).Debug("fetch")
 	ref := MakeRefKey(ctx, desc)
+
+	cw, err := ingester.Writer(ctx, ref, desc.Size, desc.Digest)
```

Comment:
Just getting that error with this change?

We need to avoid returning `ErrExists` if there is a writer in progress. An error should be returned which just indicates the resource is locked. At that point the dispatcher should just start following the status of the in progress lock. Not sure if the error being returned right now is correct.